ledger/accountdb_test.go

```diff
@@ -134,14 +134,11 @@ func randomFullAccountData(rewardsLevel, lastCreatableID uint64) (basics.Account
 				tv := basics.TealValue{
 					Type: basics.TealBytesType,
 				}
-				bytes := make([]byte, crypto.RandUint64()%512)
+				bytes := make([]byte, crypto.RandUint64()%uint64(config.MaxBytesKeyValueLen))
 				crypto.RandBytes(bytes[:])
 				tv.Bytes = string(bytes)
 				ap.KeyValue[appName] = tv
 			}
-			if len(ap.KeyValue) == 0 {
-				ap.KeyValue = nil
-			}
```

Comment:
it's ok that the code knows how to treat that in the same way now, but wouldn't that change require a protocol upgrade ? i.e. "old" binary vs. "new" binary would evaluate a TEAL code differently ?